internal/pkg/util/env/create.go

```diff
@@ -100,10 +101,21 @@ func warnDeprecatedEnvUsage(hostEnvs []string) {
 			strs := strings.SplitN(env, "=", 2)
 			if len(strs) == 2 {
 				key := strs[0][len(LegacySingularityEnvPrefix):]
+				value := strs[1]
 				if key != "" {
 					legacyEnv := LegacySingularityEnvPrefix + key
 					newEnv := ApptainerEnvPrefix + key
-					sylog.Infof("Environment variable %v is set, but %v is preferred", legacyEnv, newEnv)
+					skip := false
+					targetValue := fmt.Sprintf("%s=%s", newEnv, value)
```

Comment:
Oh now I see why it was checking the prefix, because it includes also the value.  We used to have a warning if the variables were both set to different values, because that can also be a problem for people.  The warning said the settings were different, but we were using the APPTAINER variable.  Check elsewhere for the exact wording.  So the thing to do is to check prefix, but with an `=` sign appended, and then check to see if the values are the same or different to decide whether to print the warning or info message.

Reply:
https://github.com/JasonYangShadow/apptainer/blob/332bed37c1809c7464956eda95348a7c75625468/internal/pkg/util/env/env.go#L78